### ISSUE TITLE
Grunt Cannot read property 'start' of undefined Mac OS X Yosemite - Node 0.10.29

### ISSUE THREAD

**Issue by bryanrasmussen** [butterproject/butter-desktop#178]
Doing the normal install 
Got the following at start

4848:1110/230306:ERROR:nw_shell.cc(336)] TypeError: Cannot read property 'start' of undefined
>>     at app://host/src/app/bootstrap.js:3:8
>>     at app://host/src/app/bootstrap.js:9:3
>> [4848:1110/230306:INFO:CONSOLE(3)] "Uncaught TypeError: Cannot read property 'start' of undefined", source: app://host/src/app/bootstrap.js (3)

node -v v0.10.29
Mac OS X Yosemite 10.10.5

trying to reinstall with the ./make_butter.sh script.

**Comment by pelewin** [comment]
facing the same problem

**Comment by bryanrasmussen** [comment]
reinstaling with build script didn't fix issue - didn't expect it to either
of course. I can probably look at code later today.

**Comment by bryanrasmussen** [comment]
with some cursory examination it looks like App is undefined so in
watchlist_browser.js

var WatchlistBrowser = App.View.PCTBrowser.extend({
        collectionModel: App.Model.WatchlistCollection
    });

you can't read property "View" of undefined...
when I go back through the whole terminal I can see there are a lot of is
not defined etc.

Can it be something about the app uri scheme on mac yosemite etc. that
means that various components etc. are not being loaded?

**Comment by bryanrasmussen** [comment]
I think it is definitely an issue with app uri scheme here - below

Running "exec:mac32" (exec) task
>> [4993:1110/233825:ERROR:nw_shell.cc(336)] ReferenceError: vjs is not
defined
>>     at app://host/src/app/vendor/videojshooks.js:1:1
>> [4993:1110/233825:INFO:CONSOLE(1)] "Uncaught ReferenceError: vjs is not
defined", source: app://host/src/app/vendor/videojshooks.js (1)
>> [4993:1110/233825:ERROR:nw_shell.cc(336)] ReferenceError: videojs is not
defined
>>     at app://host/src/app/vendor/videojsplugins.js:3:1
>> [4993:1110/233825:INFO:CONSOLE(3)] "Uncaught ReferenceError: videojs is
not defined", source: app://host/src/app/vendor/videojsplugins.js (3)
>> [4993:1110/233825:ERROR:nw_shell.cc(336)] ReferenceError: jQuery is not
defined
>>     at app://host/src/app/lib/jquery.plugins.js:36:4
>> [4993:1110/233825:INFO:CONSOLE(36)] "Uncaught ReferenceError: jQuery is
not defined", source: app://host/src/app/lib/jquery.plugins.js (36)
>> [4993:1110/233826:INFO:CONSOLE(6)] ""Database path:
/Users/bryan/Library/Application Support/Butter"", source:
app://host/src/app/database.js (6)
>> [4993:1110/233826:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/app.js:60:15
>> [4993:1110/233826:INFO:CONSOLE(60)] "Uncaught ReferenceError: Backbone
is not defined", source: app://host/src/app/app.js (60)
>> [4993:1110/233826:ERROR:nw_shell.cc(336)] TypeError: Cannot set property
'Config' of undefined
>>     at app://host/src/app/lib/config.js:223:16
>>     at app://host/src/app/lib/config.js:224:3
>> [4993:1110/233826:INFO:CONSOLE(223)] "Uncaught TypeError: Cannot set
property 'Config' of undefined", source: app://host/src/app/lib/config.js
(223)
>> [4993:1110/233826:ERROR:nw_shell.cc(336)] TypeError: Cannot set property
'Updater' of undefined
>>     at app://host/src/app/updater.js:412:17
>>     at app://host/src/app/updater.js:414:3
>> [4993:1110/233826:INFO:CONSOLE(412)] "Uncaught TypeError: Cannot set
property 'Updater' of undefined", source: app://host/src/app/updater.js
(412)
>> [4993:1110/233827:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'vent' of undefined
>>     at app://host/src/app/httpapi.js:755:8
>>     at app://host/src/app/httpapi.js:767:3
>> [4993:1110/233827:INFO:CONSOLE(755)] "Uncaught TypeError: Cannot read
property 'vent' of undefined", source: app://host/src/app/httpapi.js (755)
>> [4993:1110/233827:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Localization' of undefined
>>     at app://host/src/app/language.js:26:4
>> [4993:1110/233827:INFO:CONSOLE(26)] "Uncaught TypeError: Cannot read
property 'Localization' of undefined", source:
app://host/src/app/language.js (26)
>> [4993:1110/233829:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'vent' of undefined
>>     at app://host/src/app/lib/streamer.js:496:8
>>     at app://host/src/app/lib/streamer.js:501:3
>> [4993:1110/233829:INFO:CONSOLE(496)] "Uncaught TypeError: Cannot read
property 'vent' of undefined", source: app://host/src/app/lib/streamer.js
(496)
>> [4993:1110/233829:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/device/generic.js:6:18
>>     at app://host/src/app/lib/device/generic.js:163:3
>> [4993:1110/233829:INFO:CONSOLE(6)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/device/generic.js (6)
>> [4993:1110/233829:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Device' of undefined
>>     at app://host/src/app/lib/device/airplay.js:5:25
>>     at app://host/src/app/lib/device/airplay.js:53:3
>> [4993:1110/233829:INFO:CONSOLE(5)] "Uncaught TypeError: Cannot read
property 'Device' of undefined", source:
app://host/src/app/lib/device/airplay.js (5)
>> [4993:1110/233829:ERROR:nw_shell.cc(336)] Error: Cannot find module
'./lib/Browser'
>>     at Function.Module._resolveFilename (module.js:327:15)
>>     at Function.Module._load (module.js:264:25)
>>     at Module.require (module.js:356:17)
>>     at require (module.js:375:17)
>>     at Object.<anonymous>
(/Users/bryan/projects/repositories/butter-desktop/node_modules/nodecast-js/index.js:3:18)
>>     at Module._compile (module.js:451:26)
>>     at Object.Module._extensions..js (module.js:469:10)
>>     at Module.load (module.js:346:32)
>>     at Function.Module._load (module.js:301:12)
>>     at Module.require (module.js:356:17)
>> [4993:1110/233829:INFO:CONSOLE(329)] "Uncaught Error: Cannot find module
'./lib/Browser'", source: module.js (329)
>> [4993:1110/233831:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Device' of undefined
>>     at app://host/src/app/lib/device/chromecast.js:5:25
>>     at app://host/src/app/lib/device/chromecast.js:167:3
>> [4993:1110/233831:INFO:CONSOLE(5)] "Uncaught TypeError: Cannot read
property 'Device' of undefined", source:
app://host/src/app/lib/device/chromecast.js (5)
>> [4993:1110/233831:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Device' of undefined
>>     at app://host/src/app/lib/device/ext-player.js:5:25
>>     at app://host/src/app/lib/device/ext-player.js:265:3
>> [4993:1110/233831:INFO:CONSOLE(5)] "Uncaught TypeError: Cannot read
property 'Device' of undefined", source:
app://host/src/app/lib/device/ext-player.js (5)
>> [4993:1110/233831:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/subtitle/generic.js:96:21
>>     at app://host/src/app/lib/subtitle/generic.js:226:3
>> [4993:1110/233831:INFO:CONSOLE(96)] "Uncaught ReferenceError: Backbone
is not defined", source: app://host/src/app/lib/subtitle/generic.js (96)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Subtitles' of undefined
>>     at app://host/src/app/lib/subtitle/server.js:87:8
>>     at app://host/src/app/lib/subtitle/server.js:88:3
>> [4993:1110/233832:INFO:CONSOLE(87)] "Uncaught TypeError: Cannot read
property 'Subtitles' of undefined", source:
app://host/src/app/lib/subtitle/server.js (87)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/models/filter.js:4:18
>>     at app://host/src/app/lib/models/filter.js:21:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/models/filter.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/models/movie.js:4:17
>>     at app://host/src/app/lib/models/movie.js:29:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/models/movie.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Model' of undefined
>>     at app://host/src/app/lib/models/show.js:4:19
>>     at app://host/src/app/lib/models/show.js:22:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Model' of undefined", source:
app://host/src/app/lib/models/show.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/models/generic_collection.js:4:25
>>     at app://host/src/app/lib/models/generic_collection.js:159:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/models/generic_collection.js
(4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Model' of undefined
>>     at app://host/src/app/lib/models/movie_collection.js:4:30
>>     at app://host/src/app/lib/models/movie_collection.js:18:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Model' of undefined", source:
app://host/src/app/lib/models/movie_collection.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/models/stream_info.js:4:22
>>     at app://host/src/app/lib/models/stream_info.js:97:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/models/stream_info.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Model' of undefined
>>     at app://host/src/app/lib/models/show_collection.js:4:29
>>     at app://host/src/app/lib/models/show_collection.js:16:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Model' of undefined", source:
app://host/src/app/lib/models/show_collection.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Model' of undefined
>>     at app://host/src/app/lib/models/anime_collection.js:4:30
>>     at app://host/src/app/lib/models/anime_collection.js:16:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Model' of undefined", source:
app://host/src/app/lib/models/anime_collection.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Model' of undefined
>>     at app://host/src/app/lib/models/indie_collection.js:4:30
>>     at app://host/src/app/lib/models/indie_collection.js:16:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Model' of undefined", source:
app://host/src/app/lib/models/indie_collection.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/models/favorite_collection.js:4:30
>>     at app://host/src/app/lib/models/favorite_collection.js:70:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/models/favorite_collection.js
(4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Model' of undefined
>>     at app://host/src/app/lib/models/watchlist_collection.js:4:34
>>     at app://host/src/app/lib/models/watchlist_collection.js:17:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Model' of undefined", source:
app://host/src/app/lib/models/watchlist_collection.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/models/notification.js:4:24
>>     at app://host/src/app/lib/models/notification.js:44:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/models/notification.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Config' of undefined
>>     at app://host/src/app/lib/cache.js:4:20
>>     at app://host/src/app/lib/cache.js:132:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Config' of undefined", source: app://host/src/app/lib/cache.js (4)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot set property
'CacheV2' of undefined
>>     at app://host/src/app/lib/cachev2.js:132:17
>>     at app://host/src/app/lib/cachev2.js:134:3
>> [4993:1110/233832:INFO:CONSOLE(132)] "Uncaught TypeError: Cannot set
property 'CacheV2' of undefined", source: app://host/src/app/lib/cachev2.js
(132)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot set property
'ProviderTypes' of undefined
>>     at app://host/src/app/lib/providers/generic.js:6:23
>>     at app://host/src/app/lib/providers/generic.js:87:3
>> [4993:1110/233832:INFO:CONSOLE(6)] "Uncaught TypeError: Cannot set
property 'ProviderTypes' of undefined", source:
app://host/src/app/lib/providers/generic.js (6)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/cache_provider.js:47:8
>>     at app://host/src/app/lib/providers/cache_provider.js:48:3
>> [4993:1110/233832:INFO:CONSOLE(47)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/cache_provider.js (47)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/cache_providerv2.js:42:8
>>     at app://host/src/app/lib/providers/cache_providerv2.js:43:3
>> [4993:1110/233832:INFO:CONSOLE(42)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/cache_providerv2.js (42)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/trakttv.js:53:26
>>     at app://host/src/app/lib/providers/trakttv.js:717:3
>> [4993:1110/233832:INFO:CONSOLE(53)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/trakttv.js (53)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/vodo.js:15:23
>>     at app://host/src/app/lib/providers/vodo.js:206:3
>> [4993:1110/233832:INFO:CONSOLE(15)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/vodo.js (15)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/archive.js:265:8
>>     at app://host/src/app/lib/providers/archive.js:267:3
>> [4993:1110/233832:INFO:CONSOLE(265)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/archive.js (265)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/favorites.js:186:8
>>     at app://host/src/app/lib/providers/favorites.js:188:3
>> [4993:1110/233832:INFO:CONSOLE(186)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/favorites.js (186)
>> [4993:1110/233832:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/watchlist.js:4:20
>>     at app://host/src/app/lib/providers/watchlist.js:205:3
>> [4993:1110/233832:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/watchlist.js (4)
>> [4993:1110/233834:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'Providers' of undefined
>>     at app://host/src/app/lib/providers/opensubtitles.js:35:8
>>     at app://host/src/app/lib/providers/opensubtitles.js:37:3
>> [4993:1110/233834:INFO:CONSOLE(35)] "Uncaught TypeError: Cannot read
property 'Providers' of undefined", source:
app://host/src/app/lib/providers/opensubtitles.js (35)
>> [4993:1110/233834:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'settings' of undefined
>>     at app://host/src/app/lib/providers/torrent_cache.js:10:31
>>     at app://host/src/app/lib/providers/torrent_cache.js:285:3
>> [4993:1110/233834:INFO:CONSOLE(10)] "Uncaught TypeError: Cannot read
property 'settings' of undefined", source:
app://host/src/app/lib/providers/torrent_cache.js (10)
>> [4993:1110/233834:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/title_bar.js:11:20
>>     at app://host/src/app/lib/views/title_bar.js:108:3
>> [4993:1110/233834:INFO:CONSOLE(11)] "Uncaught ReferenceError: Backbone
is not defined", source: app://host/src/app/lib/views/title_bar.js (11)
>> [4993:1110/233834:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/main_window.js:6:22
>>     at app://host/src/app/lib/views/main_window.js:586:3
>> [4993:1110/233834:INFO:CONSOLE(6)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/main_window.js (6)
>> [4993:1110/233834:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/movie_detail.js:4:8
>>     at app://host/src/app/lib/views/movie_detail.js:387:3
>> [4993:1110/233834:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/movie_detail.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/show_detail.js:9:22
>>     at app://host/src/app/lib/views/show_detail.js:848:3
>> [4993:1110/233835:INFO:CONSOLE(9)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/show_detail.js (9)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/settings_container.js:9:20
>>     at app://host/src/app/lib/views/settings_container.js:765:3
>> [4993:1110/233835:INFO:CONSOLE(9)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/settings_container.js (9)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/init_modal.js:4:21
>>     at app://host/src/app/lib/views/init_modal.js:64:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/init_modal.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/disclaimer.js:4:27
>>     at app://host/src/app/lib/views/disclaimer.js:33:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/disclaimer.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/about.js:4:17
>>     at app://host/src/app/lib/views/about.js:67:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/about.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/torrent_collection.js:8:29
>>     at app://host/src/app/lib/views/torrent_collection.js:417:3
>> [4993:1110/233835:INFO:CONSOLE(8)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/torrent_collection.js (8)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/file_selector.js:7:24
>>     at app://host/src/app/lib/views/file_selector.js:201:3
>> [4993:1110/233835:INFO:CONSOLE(7)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/file_selector.js (7)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/help.js:5:16
>>     at app://host/src/app/lib/views/help.js:52:3
>> [4993:1110/233835:INFO:CONSOLE(5)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/help.js (5)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/issue.js:8:17
>>     at app://host/src/app/lib/views/issue.js:270:3
>> [4993:1110/233835:INFO:CONSOLE(8)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/issue.js (8)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/keyboard.js:4:20
>>     at app://host/src/app/lib/views/keyboard.js:29:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/keyboard.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/notification.js:4:8
>>     at app://host/src/app/lib/views/notification.js:50:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/notification.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/player/loading.js:4:19
>>     at app://host/src/app/lib/views/player/loading.js:281:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/player/loading.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/player/player.js:13:18
>>     at app://host/src/app/lib/views/player/player.js:922:3
>> [4993:1110/233835:INFO:CONSOLE(13)] "Uncaught ReferenceError: Backbone
is not defined", source: app://host/src/app/lib/views/player/player.js (13)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/browser/generic_browser.js:12:22
>>     at app://host/src/app/lib/views/browser/generic_browser.js:137:3
>> [4993:1110/233835:INFO:CONSOLE(12)] "Uncaught ReferenceError: Backbone
is not defined", source:
app://host/src/app/lib/views/browser/generic_browser.js (12)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/browser/movie_browser.js:4:27
>>     at app://host/src/app/lib/views/browser/movie_browser.js:13:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/browser/movie_browser.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/browser/filter_bar.js:5:8
>>     at app://host/src/app/lib/views/browser/filter_bar.js:476:3
>> [4993:1110/233835:INFO:CONSOLE(5)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/browser/filter_bar.js (5)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/browser/item.js:7:16
>>     at app://host/src/app/lib/views/browser/item.js:476:3
>> [4993:1110/233835:INFO:CONSOLE(7)] "Uncaught ReferenceError: Backbone is
not defined", source: app://host/src/app/lib/views/browser/item.js (7)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] ReferenceError: Backbone is
not defined
>>     at app://host/src/app/lib/views/browser/list.js:24:21
>>     at app://host/src/app/lib/views/browser/list.js:507:3
>> [4993:1110/233835:INFO:CONSOLE(24)] "Uncaught ReferenceError: Backbone
is not defined", source: app://host/src/app/lib/views/browser/list.js (24)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/browser/show_browser.js:4:26
>>     at app://host/src/app/lib/views/browser/show_browser.js:13:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/browser/show_browser.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/browser/anime_browser.js:4:27
>>     at app://host/src/app/lib/views/browser/anime_browser.js:14:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/browser/anime_browser.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/browser/indie_browser.js:4:27
>>     at app://host/src/app/lib/views/browser/indie_browser.js:14:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/browser/indie_browser.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/browser/favorite_browser.js:4:30
>>     at app://host/src/app/lib/views/browser/favorite_browser.js:13:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/browser/favorite_browser.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'View' of undefined
>>     at app://host/src/app/lib/views/browser/watchlist_browser.js:4:31
>>     at app://host/src/app/lib/views/browser/watchlist_browser.js:9:3
>> [4993:1110/233835:INFO:CONSOLE(4)] "Uncaught TypeError: Cannot read
property 'View' of undefined", source:
app://host/src/app/lib/views/browser/watchlist_browser.js (4)
>> [4993:1110/233835:ERROR:nw_shell.cc(336)] TypeError: Cannot read
property 'start' of undefined
>>     at app://host/src/app/bootstrap.js:3:8
>>     at app://host/src/app/bootstrap.js:9:3
>> [4993:1110/233835:INFO:CONSOLE(3)] "Uncaught TypeError: Cannot read
property 'start' of undefined", source: app://host/src/app/bootstrap.js (3)


On Wed, Nov 11, 2015 at 11:43 AM, bryan rasmussen <rasmussen.bryan@gmail.com
> wrote:

> with some cursory examination it looks like App is undefined so in
> watchlist_browser.js
>
> var WatchlistBrowser = App.View.PCTBrowser.extend({
>         collectionModel: App.Model.WatchlistCollection
>     });
>
> you can't read property "View" of undefined...
> when I go back through the whole terminal I can see there are a lot of is
> not defined etc.
>
> Can it be something about the app uri scheme on mac yosemite etc. that
> means that various components etc. are not being loaded?
>
>
>
> On Wed, Nov 11, 2015 at 8:29 AM, bryan rasmussen <

**Comment by bryanrasmussen** [comment]
I checked /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/Laun\
chServices.framework/Versions/A/Support/lsregister -dump | open -fe and it turns out the app uri scheme isn't registered - probably there should be a check somewhere  and if it doesn't work tell them to register the handler. 

Now the question is if anyone knows an implementation of the protocol that runs on mac?

**Comment by bryanrasmussen** [comment]
So anyway, can anyone recommend an implementation of the app protocol - in node.js or for mac (that is to say I don't need it to be in node.js but I do need it to work for mac, but I would prefer it were implemented in node.js and working for mac)
I suppose since there are app: uris all over the place that butter expects it to be installed and working - can anyone tell me what implementation there should be of the app protocol that fits this expectation?

there doesn't seem to be anything that installs an implementation? 

I've found this one 
https://github.com/nwjs/nw.js/wiki/App-protocol
but that isn't in the build. 

Does butter have its own implementation somewhere - if so can anyone point me where in the source it is. I don't have any bindings for the uri scheme on my computer - is there a script that should run to generate the bindings? Just wondering because it seems like the program is completely written with the base assumption that this relatively new and very useful looking protocol is just running out of the box on my OS - since, as I said, it looks useful I would like to find out where that assumption comes from?

**Comment by bryanrasmussen** [comment]
aw of course looked back and nw is required, so I guess that answers where the app protocol is supposed to be implemented at - but is there supposed to be any register for the protocol anywhere - also IIRC the method for uri scheme registration got more convoluted recently - prob. with Yosemite which would explain the issue. Does anyone know if NW is supposed to register the scheme automatically - if Butter is supposed to do it or otherwise?

**Comment by bryanrasmussen** [comment]
well I got nw installed but no help. One thing I noted that was wrong aside from all this was that Butter evidently thinks it should run exec:mac32 but it should run exec:mac64. I changed the gruntfile to force mac64 and rebuilt (rebuilding had ENOTEMPTY errors because it couldn't rmdir of the original builds because there was content in them (DS_Store files) 

anyway that's about as far as I go. Maybe I should try to install on Ubuntu instead.

**Comment by bryanrasmussen** [comment]
Finally worked. What did I have to do - I had to install nwjs set up the grunt task to only run nwjs, I tried to build a couple times but that didn't work out. Had to remove the 64 binary and folders because there's a rmdir command and won't run as per the comment I made above. 

So finally I ran ./make_butter.sh, removing folders first as described, and then it said there wasn't any grunt command anymore. So then I installed grunt again, and then did grunt start and it took a 3-4 minutes to start but it worked.
